src/pages/InputDataPage.jsx

```diff
@@ -0,0 +1,182 @@
+/* eslint-disable react/prop-types */
+import { useState, useEffect } from 'react';
+import { fetchJoinedEventsById } from '../utils/fuseUtils';
+import { getEventById } from '../utils/eventsUtils';
+import Backend from '../utils/utils';
+import Fuse from 'fuse.js';
+import VolunteerEventsTable from '../components/Checkin/VolunteerEventsTable';
+import CheckinStatsDashboard from '../components/Checkin/CheckinStatsDashboard';
+
+import { useParams } from 'react-router-dom';
+import { Container, Flex, Box, Input, InputGroup, InputLeftElement, Alert } from '@chakra-ui/react';
+
+import { GreyCustomSearchIcon } from '../components/Icons/CustomSearchIcon';
+
+const InputDataPage = () => {
+  const [joinedData, setJoinedData] = useState([]);
+  const [volunteerResults, setVolunteerResults] = useState([]);
+  const [input, setInput] = useState('');
+  const [event, setEvent] = useState('');
+  const [registered, setRegistered] = useState('');
+  const [checkin, setCheckin] = useState('');
+  const { eventId } = useParams();
+  const [displayedVolunteers, setDisplayedVolunteers] = useState([]);
+
+  useEffect(() => {
+    // 0 is all, 1 is checked-in, 2 is not checked-in, 3 are guests
+    setDisplayedVolunteers(volunteerResults.filter(volunteer => volunteer.is_checked_in === true));
+  }, [volunteerResults]);
+
+  const setData = async () => {
+    try {
+      const data = await fetchJoinedEventsById(eventId); // joined data for table rendering
+      const event = await getEventById(eventId); // event data for page rendering eg. image src
+      //   console.log(event);
+      setJoinedData(data);
+      setEvent(event);
+      getRegistered(event.id);
+      getCheckin(event.id);
+    } catch (error) {
+      console.error('Error fetching data:', error);
+    }
+  };
+
+  /*
+    useEffect for getting joined data on first render
+  */
+  useEffect(() => {
+    setData();
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
+
+  /*
+    This useffect is used for updating the display volunteer data based on user input utilizing fuzzy search
+  */
+  useEffect(() => {
+    // If input is empty, display all volunteers, else conduct the search
+    if (input.trim() === '') {
+      setVolunteerResults(joinedData);
+    } else {
+      const options = {
+        keys: ['first_name', 'last_name', 'email'],
+      };
+      const fuse = new Fuse(
+        joinedData.filter(item => item.event_id == eventId || eventId == -1),
+        options,
+      );
+      const searchResult = fuse.search(input);
+      const reduceResult = searchResult.map(result => result.item);
+      setVolunteerResults(reduceResult);
+    }
+  }, [input, joinedData]);
```

Comment:
⚠️ **[eslint]** <[react-hooks/exhaustive-deps](https://github.com/facebook/react/issues/14920)> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>React Hook useEffect has a missing dependency: 'eventId'. Either include it or remove the dependency array.